edge-hub/test/Microsoft.Azure.Devices.Edge.Hub.CloudProxy.Test/ConnectivityAwareClientTest.cs

```diff
@@ -37,6 +37,14 @@ void ConnectionStatusChangedHandler(ConnectionStatus status, ConnectionStatusCha
             deviceConnectivityManager.InvokeDeviceConnected();
             deviceConnectivityManager.InvokeDeviceDisconnected();
 
+            // Assert
+            Assert.Equal(0, connectionStatusChangedHandlerCount);
```

Comment:
I think this should be checked in a different test that it doesn't get called if OpenAsync was not called

Reply:
Yeah, that was my intention to check that this hasn't gotten called if OpenAsync is not called. Are you suggesting I write a separate test for that?

Reply:
I think a separate test would be good, if this test fails is because of CloseAsync bug, not because of other issues

---
In reply to: [304158137](https://github.com/Azure/iotedge/pull/1412#discussion_r304158137) [](ancestors = 304158137)

Reply:
I actually have separate tests for Close and Open. I think I understand what you are saying.. but we haven't really followed this pattern in other cases (where we test only the case in consideration). Maybe it is a good pattern, but increases the number of tests needed, I guess. 
I will keep it like this for now. Lmk if you feel strongly about it, I will remove the asserts.